lib/Dialect/Ibis/Transforms/IbisContainerize.cpp

```diff
@@ -0,0 +1,134 @@
+//===- IbisContainerize.cpp - Implementation of containerizing ------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "PassDetails.h"
+
+#include "circt/Dialect/Ibis/IbisDialect.h"
+#include "circt/Dialect/Ibis/IbisOps.h"
+#include "circt/Dialect/Ibis/IbisPasses.h"
+#include "circt/Dialect/Ibis/IbisTypes.h"
+
+#include "circt/Support/SymCache.h"
+#include "mlir/Transforms/DialectConversion.h"
+
+using namespace circt;
+using namespace ibis;
+
+// Iterates the symbolcache until a unique name is found.
+static StringAttr getUniqueName(mlir::MLIRContext *ctx, StringAttr baseName,
```

Comment:
This feels like something which should be shared with other passes... (not for this PR.)

Reply:
maybe a TODO is appropriate for now.